src/main/java/io/aiven/kafka/connect/transforms/ExtractTimestamp.java

```diff
@@ -70,9 +70,9 @@ public R apply(final R record) {
 
         final long newTimestamp;
         if (fieldValue instanceof Long) {
-            newTimestamp = (long) fieldValue;
+            newTimestamp = config.timestampResolution().convert((long) fieldValue);
         } else if (fieldValue instanceof Date) {
-            newTimestamp = ((Date) fieldValue).getTime();
+            newTimestamp = config.timestampResolution().convert(((Date) fieldValue).getTime());
```

Comment:
Should it be converted for `Date`?

Reply:
yes. `fieldValue` is `java.lang.Object`

Reply:
Sorry, I mean: should resolution conversion be applied?

Reply:
Yes: quote from docs to `org.apache.kafka.connect.data.Timestamp`
```java
 *  A timestamp representing an absolute time, without timezone information. The corresponding Java type is a
 *  java.util.Date. The underlying representation is a long representing the number of milliseconds since Unix epoch.
```

Reply:
Let's say `fieldValue` equals `Date(2020-11-17T07:57:40.140Z)`. What the expected output should be?

Reply:
If you mean time zones then `org.apache.kafka.connect.data.Timestamp` doesn't support them.

Reply:
((Date) fieldValue).getTime() is always milliseconds so no conversion needed.

Reply:
Yep, please use this test as a reference:

```java
    @Test
    void structWithTimestampFieldInSeconds() {
        final Schema schema = SchemaBuilder.struct()
                .field(FIELD, Timestamp.SCHEMA)
                .build();

        final ZonedDateTime date = ZonedDateTime.of(
            2020, 11, 15, 1, 2, 3, 4, ZoneId.of("UTC"));
        final Instant instant = date.toInstant();

        final var props = Map.of(
            "timestamp.resolution", "seconds"
        );
        final SinkRecord originalRecord = record(null, new Struct(schema).put(FIELD, Date.from(instant)));
        final SinkRecord transformedRecord = transformation(props).apply(originalRecord);
        
        assertEquals(
                setNewTimestamp(originalRecord, instant.toEpochMilli()),
                transformedRecord);
    }
```

Reply:
(also it's worth having a similar test for Map).